test/basic.test.js

```diff
@@ -9,6 +9,36 @@ const Strategy = require('passport-http-bearer').Strategy
 
 const expressPlugin = require('../index')
 
+test('Readme Example', t => {
+  t.plan(2)
+  const fastify = Fastify()
+  t.teardown(fastify.close)
+  const router = require('express').Router()
+
+  router.use(function (req, res, next) {
+    res.setHeader('x-custom', true)
+    next()
+  })
+
+  router.get('/hello', (req, res) => {
+    res.status(201)
+    res.json({ hello: 'world' })
+  })
+
+  router.get('/foo', (req, res) => {
+    res.status(400)
+    res.json({ foo: 'bar' })
+  })
+
+  fastify.register(expressPlugin)
+    .after(() => fastify.use(router))
```

Comment:
```suggestion
    .after(() => { fastify.use(router) })
```

I think you are creating a loop between promises. `fastify` is now a thenable.

I'll recommend you to use `await` anyway to avoid those hiccups.